contributing/development/cpp_usage_guidelines.rst

```diff
@@ -100,6 +100,12 @@ pull request.
 To follow the existing style, please use standard ``#ifdef``-based include
 guards instead of ``#pragma once`` in new files.
 
+``try``-``catch`` blocks
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+C++ style exception handling using ``try`` and ``catch`` blocks is forbidden.
```

Comment:
```suggestion
C++ style exception handling using ``try`` and ``catch`` blocks is forbidden.
This restriction is in place for several reasons, including performance, binary
size and code complexity.
Use :ref:`doc_common_engine_methods_and_macros_error_macros` instead.
```

Also remember to add this in `contributing/development/core_and_modules/common_engine_methods_and_macros.rst` above the `Error macros` heading:

```
.. _doc_common_engine_methods_and_macros_error_macros:
```